ethcore/src/private_transactions/private_transactions.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use transaction::UnverifiedTransaction;
+use error::Error;
+use parking_lot::RwLock;
+
+/// Storage for private transactions
+pub struct PrivateTransactions {
+	transactions: RwLock<Vec<UnverifiedTransaction>>,
+	signed_transactions:  RwLock<Vec<UnverifiedTransaction>>,
```

Comment:
I'm a little confused why it needs locks here. Isn't it protected by a mutex within the `Provider`?